bond/commands/upgrade.py

```diff
@@ -20,14 +20,15 @@ def get_branch_string(branch, target):
         return branch.replace("/", "-")
 
 
-def get_latest_version(target, branch):
+def get_s3_version(target, branch, depth=0):
     url = f"https://s3.amazonaws.com/bond-updates/v2/{target}/{branch}/versions_internal.json"
     rsp = requests.get(url)
     if rsp.status_code != 200:
         raise SystemExit(
             "Failed to find an upgrade for target %s on branch %s" % (target, branch)
         )
-    return json.loads(rsp.content)["versions"][0]
+    index = depth or 0
```

Comment:
No need for `depth or 0` here, as the default argument provides the same guarantee. You're also coercing to 0 in the code that calls this function. Perhaps it best just to move all of these into the argument spec, as I described in another comment.